metadata-ingestion/src/datahub/ingestion/source/sql/mariadb.py

```diff
@@ -7,14 +15,112 @@
     support_status,
 )
 from datahub.ingestion.source.sql.mysql import MySQLConfig, MySQLSource
+from datahub.ingestion.source.sql.stored_procedures.base import (
+    BaseProcedure,
+)
+
+# MariaDB inherits shared stored procedure config from MySQL via StoredProcedureConfigMixin
+
+logger: logging.Logger = logging.getLogger(__name__)
+
+STORED_PROCEDURES_QUERY = """
+SELECT
+    ROUTINE_SCHEMA,
+    ROUTINE_NAME,
+    ROUTINE_DEFINITION,
+    ROUTINE_COMMENT,
+    CREATED,
+    LAST_ALTERED,
+    SQL_DATA_ACCESS,
+    SECURITY_TYPE,
+    DEFINER
+FROM information_schema.ROUTINES
+WHERE ROUTINE_TYPE = 'PROCEDURE'
+AND ROUTINE_SCHEMA = :schema
+"""
+
+
+class MariaDBConfig(MySQLConfig):
+    host_port: str = Field(default="localhost:3306", description="MariaDB host URL.")
+    # MariaDB inherits stored procedure configuration from MySQLConfig
+    # This includes include_stored_procedures and procedure_pattern
 
 
 @platform_name("MariaDB")
-@config_class(MySQLConfig)
+@config_class(MariaDBConfig)
 @support_status(SupportStatus.CERTIFIED)
 @capability(SourceCapability.PLATFORM_INSTANCE, "Enabled by default")
 @capability(SourceCapability.DOMAINS, "Supported via the `domain` config field")
 @capability(SourceCapability.DATA_PROFILING, "Optionally enabled via configuration")
 class MariaDBSource(MySQLSource):
-    def get_platform(self):
+    def get_platform(self) -> str:
         return "mariadb"
+
+    def _get_stored_procedures(
+        self,
+        conn: Connection,
+        db_name: str,
+        schema: str,
+    ) -> List[BaseProcedure]:
+        query = text(STORED_PROCEDURES_QUERY)
+
+        procedures_list = []
+
+        for row in conn.execute(query, {"schema": schema}):
+            # Convert SQLAlchemy Row to dict for easier and safer access
+            row_dict = dict(row)
+            procedure_name = row_dict.get("ROUTINE_NAME", "unknown")
```

Comment:
Is set but never used in error reporting since we immediately check for None and skip